src/lib/installation/widgets/ntp_server.rb

```diff
@@ -0,0 +1,143 @@
+      def label
+        # TRANSLATORS: input field label
+        _("N&TP Servers (comma or space separated)")
+      end
+
+      # Store the value of the input field if validates
+      def store
+        if servers.empty?
+          # we need to reset the previous settings after going back
+          Yast::NtpClient.ntp_selected = false
+          Yast::NtpClient.modified = false
+
+          return
+        end
+
+        Yast::NtpClient.ntp_selected = true
+        Yast::NtpClient.modified = true
+        Yast::NtpClient.ntp_conf.clear_pools
+        servers.each { |server| Yast::NtpClient.ntp_conf.add_pool(server) }
+        # run NTP as a service (not via cron)
+        Yast::NtpClient.run_service = true
+        Yast::NtpClient.synchronize_time = false
+      end
+
+      # Initializes the widget's value
+      def init
+        saved_servers = (role && role["ntp_servers"]) || default_servers
+        self.value = saved_servers.join(" ")
+      end
+
+      NOT_VALID_SERVERS_MESSAGE = N_("Not valid location for the NTP servers:\n%{servers}" \
+        "\n\nPlease, enter a valid IP or Hostname").freeze
+      # Validate input
+      #
+      # * All specified IPs or hostnames should be valid
+      # * If no server is specified, ask the user whether proceed with installation or not
+      #
+      # @return [Boolean] true if value is valid; false otherwise.
+      def validate
+        return skip_ntp_server? if servers.empty?
+        invalid_servers = servers.reject { |v| Yast::IP.Check(v) || Yast::Hostname.CheckFQ(v) }
+        return true if invalid_servers.empty?
+        Yast::Popup.Error(
+          format(_(NOT_VALID_SERVERS_MESSAGE), servers: invalid_servers.join(", "))
+        )
+
+        false
+      end
+
+      def help
+        # TRANSLATORS: a help text for the NTP server input field
+        _("<h3>NTP Servers</h3>") +
+          # TRANSLATORS: a help text for the NTP server input field
+          _("<p>Enter the host name or the IP address of the NTP server which will be used for " \
+            "synchronizing the time on this machine</p>") +
+          # TRANSLATORS: a help text for the NTP server input field
+          _("<p>Use comma (,) or space to separate multiple values.</p>")
+      end
+
+    private
+
+      # Parse the widget's value an return the potential list of hostnames/addresses
+      #
+      # @return [Array<String>] List of hostnames/addresses
+      def servers
+        value.to_s.tr(",", " ").split(" ")
+      end
+
+      # Check if the user wants to intentionally skip the NTP server configuration
+      #
+      # @return [Boolean] true if user wants to skip it; false otherwise.
+      def skip_ntp_server?
+        Yast::Popup.AnyQuestion(
+          _("NTP Servers"),
+          # TRANSLATORS: error message for invalid ntp server name/address
+          _("You have not configured an NTP server. This may lead to\n" \
+          "your cluster not functioning properly or at all.\n" \
```

Comment:
I am not sure about wording if it is appropriate for any possible opensuse use case.

Reply:
Since this is initially moved from `yast-caasp` to be used by Kubic, do you think we should take care of this right now or it could be postponed until it will be used in other use case? Just asking, I have not strong opinion about when we should address it.

Reply:
well, add there at least TODO: so we are aware of it.

Reply:
Ah, good point, this dialog was originally designed for Kubic/CaaSP. For generic usage we need some more generic message.

Reply:
Something like 
```suggestion
          "your system not functioning properly.\n" \
```
???